src/smolagents/models.py

```diff
@@ -1161,6 +1170,136 @@ def create_client(self):
         return openai.AzureOpenAI(**self.client_kwargs)
 
 
+class AmazonBedrockServerModel(ApiModel):
+    """
+    A model class for interacting with Amazon Bedrock Server models through the Bedrock API.
+
+    This class provides an interface to interact with various Bedrock language models,
+    allowing for customized model inference, guardrail configuration, message handling,
+    and other parameters allowed by boto3 API.
+
+    Parameters:
+        model_id (`str`):
+            The model identifier to use on Bedrock (e.g. "us.amazon.nova-pro-v1:0").
+        boto3_client (`boto3.client`, *optional*):
+            A custom boto3 client for AWS interactions. If not provided, a default client will be created.
+        inference_config (`dict`, *optional*):
+            Configuration parameters for model inference, such as temperature, max tokens, and other model-specific settings.
+        guardrail_config (`dict`, *optional*):
+            Configuration for model guardrails to control and moderate model behavior and outputs.
+        api_kwargs (`dict`, *optional*):
+            Additional keyword arguments to be passed to boto3 bedrock-runtime API requests.
+        custom_role_conversions (`dict[str, str]`, *optional*):
+            Custom role conversion mapping to convert message roles in others.
+            Useful for specific models that do not support specific message roles like "system".
+        flatten_messages_as_text (`bool`, default `False`):
+            Whether to flatten messages as text.
+
+    Example:
+        >>> bedrock_model = AmazonBedrockServerModel(
+        ...     model_id='us.amazon.nova-pro-v1:0',
+        ...     inference_config={'temperature': 0.7, 'max_tokens': 3000}
+        ... )
+    """
+
+    def __init__(
+        self,
+        model_id: str,
+        boto3_client=None,
+        inference_config: dict[str, Any] | None = None,
+        guardrail_config: dict[str, Any] | None = None,
+        bedrock_api_kwargs: dict[str, Any] | None = None,
+        custom_role_conversions: dict[str, str] | None = None,
+        flatten_messages_as_text: bool = False,
+        **kwargs,
+    ):
+        self.model_id = model_id
+        self.client = boto3_client or self.create_client()
+        self.inference_config = inference_config
+        self.guardrail_config = guardrail_config
+        self.bedrock_api_kwargs = bedrock_api_kwargs
+        self.custom_role_conversions = custom_role_conversions
```

Comment:
I think you could set here the model-specific role conversions (that you implemented below in `__call__`): I think it makes more sense to set them in the `__init__` method instead.

Reply:
Agree, changed!